French translation

GiGiDKR opened the pull request:
This pull request introduces French language support across various parts of the project, including documentation, prompts, and error messages. The most important changes include updates to multiple `README` files, the addition of a French version of `options.md`, and updates to the internationalization files.

Language support additions:

* `README.md`, `README_zh-Hans.md`: Added French language link and icon. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L13-R13) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L93-R93) [[3]](diffhunk://#diff-9dc475598a564cbae2b0fd91890420930901991f7855480c2e26a1b67ab33280L13-R13) [[4]](diffhunk://#diff-9dc475598a564cbae2b0fd91890420930901991f7855480c2e26a1b67ab33280L108-R108)
* `docs/options.md`, `docs/options_zh-Hans.md`: Added French language link and icon. [[1]](diffhunk://#diff-2e046795c81fb2bcc2667705f88c6421fa988047f147ddb2333ec8da5f638b7bR5) [[2]](diffhunk://#diff-2e046795c81fb2bcc2667705f88c6421fa988047f147ddb2333ec8da5f638b7bR138) [[3]](diffhunk://#diff-48136d3d8dd1e4cb9e1b3a7c0c21c0a3fd086346a450ad0665b54361cb9cd7e1R5) [[4]](diffhunk://#diff-48136d3d8dd1e4cb9e1b3a7c0c21c0a3fd086346a450ad0665b54361cb9cd7e1R144)
* [`docs/options_fr.md`](diffhunk://#diff-bc981b39743a232b9a98ee645fd7ee39253be7c749e66c0481f8afa21c863ea4R1-R138): Added a new French version of the options documentation.

French translations:

* [`src-tauri/migration/src/m20240101_100002_seed_prompts.rs`](diffhunk://#diff-582d91deb0afd96d6b130d0387269ccf148dbf5de3ccbb147b45b941a21ebfeeL12-R13): Translated prompt content to French.
* [`src/i18n/locales/en/generic.json`](diffhunk://#diff-cf7ad3fa3a4d7a170f1ee9d99dc3a6b3710a25b9d26fa2a8a31fcc789fcebbd3L44-R45): Added French language option in the language selection.
* [`src/i18n/locales/fr/error.json`](diffhunk://#diff-b9e3c7065680a70bd5d2253e0f3d9c571827f07f760fa82c97a3624d90290971R1-R12): Added French translations for error messages.

0xfrankz commented on src/pages/Conversation.tsx:
Maybe you can move the hard coded message into i18n files (my fault for leaving them hard coded). Otherwise, leave it unchanged so I can fix it some other day.

0xfrankz commented on src-tauri/migration/src/m20240101_100002_seed_prompts.rs:
I'd like to keep this default prompt template in English. Maybe you can add a "COSTAR FR" template for the French version?

0xfrankz commented on src/pages/Settings.tsx:
Not sure when the Card tag is removed, is it necessary? Maybe we can add French by just adding the following:
```
<SelectItem value="fr">
  {t('generic:select:language-fr')}
</SelectItem>
```

0xfrankz reviewed (changes requested):
Thanks for the pull request. Do you mind change it a little bit? You can check out my comments. Merci~

GiGiDKR commented:
Added translations for error messages related to invalid conversation IDs:

- Added translation keys to local files (fr, en, zh-Hans)
- Updated Conversation.tsx to use the i18n system
- Confirmed the existence of French prompt templates [COSTAR FR, Zero-shot COT FR] (**the problem is that they are integrated into the interface regardless of the language setting... perhaps we should proceed differently?**)

0xfrankz commented:
Merged. Merci beaucoup
